### PR TITLE
[ISSUE-371] Recover volume operation tests

### LINKED ISSUE #371: Code refactoring after reservation logic change
There are multiple places marked in the code that should be refactored or improved after reservation logic change.

### PR DESCRIPTION
**Pull request by rkiselenko**
### Resolves #371 

Commented tests must be uncomment and fixed 

## PR checklist
- [x] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [x] New unit tests added
- [x] Modified code has meaningful comments
- [x] All TODOs are linked with the issues
- [x] All comments are resolved

## Testing
_Provide test details_

### CODE REVIEW

File: pkg/common/volume_operations_test.go [hunk 1]
```diff
@@ -221,42 +243,36 @@ func Test_handleVolumeInProgress(t *testing.T) {
...
-	})
+		Location:     testLVG.Spec.Name,
+	}
+
+	ctx := context.WithValue(testCtx, util.VolumeInfoKey, &util.VolumeInfo{Name: volumeID, Namespace: testNS})
+	createdVolume, err := svc.CreateVolume(ctx, tv)
 	assert.Nil(t, err)
 	assert.NotNil(t, createdVolume)
-	assert.Equal(t, expectedVolume, *createdVolume)
+	assert.Equal(t, expectedVolume, createdVolume)
```
Comment (safronovD) [inline pkg/common/volume_operations_test.go:262]: Is `assert.Equal` compare pointers by value by default?

File: pkg/common/volume_operations_test.go [hunk 2]
```diff
@@ -575,61 +617,75 @@ func TestVolumeOperationsImpl_ExpandVolume_DifferentStatuses(t *testing.T) {
...
-	svc = setupVOOperationsTest(t)
-	svc.cache.Set(volumeCR.Spec.Id, testNS)
+	svc.cache.Set(volumeCR.Spec.Id, volumeCR.Namespace)
 	volumeCR.Spec.CSIStatus = apiV1.Failed
-	err = svc.k8sClient.CreateCR(testCtx, volumeCR.Spec.Id, &volumeCR)
-	volAC := testAC2
+	err = svc.k8sClient.CreateCR(testCtx, volumeCR.Spec.Id, volumeCR)
+	volAC := &accrd.AvailableCapacity{
+		TypeMeta:   k8smetav1.TypeMeta{Kind: "AvailableCapacity", APIVersion: apiV1.APIV1Version},
+		ObjectMeta: k8smetav1.ObjectMeta{Name: uuid.New().String(), Namespace: testNS},
```
Comment (safronovD) [inline pkg/common/volume_operations_test.go:681]: AC is cluster-scoped. Do we need to set NS here?

File: pkg/common/volume_operations_test.go [hunk 3]
```diff
@@ -575,61 +617,75 @@ func TestVolumeOperationsImpl_ExpandVolume_DifferentStatuses(t *testing.T) {
...
 	var (
 		svc      *VolumeOperationsImpl
-		err      error
 		capacity = int64(util.TBYTE)
 	)
 
 	volumeCR := testVolume1
 	svc = setupVOOperationsTest(t)
-	err = svc.k8sClient.CreateCR(testCtx, volumeCR.Spec.Id, &volumeCR)
+	volumeCR.ObjectMeta.CreationTimestamp = k8smetav1.Time{}
```
Comment (safronovD) [inline pkg/common/volume_operations_test.go:625]: Isn't better to use `volumeCR := testVolume1.DeepCopy()` everywhere despite of returning overridden fields back?

File: pkg/common/volume_operations_test.go (hunk 1 above)
Comment (rkiselenko) [reply pkg/common/volume_operations_test.go:262]: I dont get.

File: pkg/common/volume_operations_test.go (hunk 2 above)
Comment (rkiselenko) [reply pkg/common/volume_operations_test.go:681]: 👍

File: pkg/common/volume_operations_test.go (hunk 3 above)
Comment (rkiselenko) [reply pkg/common/volume_operations_test.go:625]: 👍

File: pkg/common/volume_operations_test.go (hunk 1 above)
Comment (safronovD) [reply pkg/common/volume_operations_test.go:262]: Got it, thank you!

**Review by safronovD — APPROVED** [review]

**Review by mishoyama — APPROVED** [review]

**Pull request merged**